### ISSUE TITLE
SCSS is silly, bootswatch no likey

### ISSUE THREAD

**Issue by rye** [kotct/docket-old#27]
Because we're lazy for some reason (that's right, we're humans :/), I need to move this to a new issue.

It seems that the bootswatch-rails things aren't being put into what I'll call the "load path" of scss. That means that important anything starting with `bootswatch/` fails, because it can't find it.

This needs to be fixed in order to add themes. Silly SCSS.

**Comment by Sammidysam** [comment]
This is where you open an issue for the gem.

**Comment by rye** [comment]
I assume the closest repository in ambiguous situations like I was in. It's still an issue here, so the issue remains open.

**Comment by Sammidysam** [comment]
So you're saying that we should not ask the gem author for help with _his_ gem?

**Comment by rye** [comment]
No, I'm saying that I was not directed to submit an issue on `bootswatch-rails`. I was directed to "open an issue," what was ambiguous. If you look at [bootswatch-rails](https://github.com/maxim/bootswatch-rails/issues), you will see my issue.

**Comment by Sammidysam** [comment]
Oh, whoops.  I really failed with being specific.  Sorry.  We can keep this issue.  It is better than discussing on the other issue because this problem doesn't really pertain as much.

**Comment by cg505** [comment]
Here are the problems:
1. Gem does not work with Bootstrap 3.
2. Gem does not work with Rails 4.

We should just manually translate the less for the bootswatches to sass (or maybe find a translator (I think that bootswatch-rails may have on in the repo)).

**Comment by Sammidysam** [comment]
Ouch, that gem seriously needs work.  The converter is [this file](https://github.com/maxim/bootswatch-rails/blob/master/converter).  May be worth looking into.

**Comment by rye** [comment]
Or, perhaps we could just use [this thing](https://github.com/log0ymxm/bootswatch-scss) and manually copy the SCSS into our project.

The other alternative is simply to download the raw bootswatch CSS and use _that_. That's what I would do in this situation.

**Comment by Sammidysam** [comment]
The first option is probably worth trying.  The second is not ideal, but could work in the end.  It is not ideal because using SCSS will make it easier to customize.  Variables are especially helpful with this.

**Comment by rye** [comment]
I would say overrides can take care of any things that we feel the need to
change, but whatever.

**Comment by Sammidysam** [comment]
Ah, okay.  I have not dealt with CSS much, so I do not really know much.  I guess @cg505's and your opinions will matter most.

**Comment by cg505** [comment]
Why I'm the world doesn't that file use `def`? Sorry, back to the actual issue 

@four04: remember when we were developing the website and we had to manually gsub all the blue in Cyborg to #fff200? The point of sass is to make it so that we don't have to do that. We just redefine the variable that has the blue color.

I would like to try to translate the less for Yeti to sass. If that doesn't work(/is too hard), we have two options
1. Use default bootstrap.
2. Transfer entire project to less.

In all honesty, j don't know if any of those three things would really be all that hard.

**Comment by cg505** [comment]
And, as I've stated before, _the default bootstrap really is not ugly_. It was more ugly v2, but I actually really like bootstrap 3.

**Comment by Sammidysam** [comment]
Yeah, I don't have any problem using the default bootstrap.  We can always customize it further to give our application a unique look.

**Comment by rye** [comment]
I agree; not much customization needs to be done, however some _does_ need to be done. I think the general standard that we'd be going for (a pattern I see a lot on the properly-designed Internet) is a darker navbar. We can talk about that later.

@cg505 I get the whole reason to use variables, and it makes sense (hardcoding stuff is bad). I think the `less` idea sounds _much_ better, considering the fact that the [actual bootswatch](http://bootswatch.com/yeti/) (Yeti was linked as the example, but it's the same for all of the other swatches) provides a download for both of the different `.less` files that one would need, there's no need for extraneous gems or any nonsense like that.

**Comment by cg505** [comment]
Also, bootstrap itself is written in less.

**Comment by rye** [comment]
So we should use `less` instead?

**Comment by cg505** [comment]
The only thing that I would worry about is the fact that most rails things are done in sass, so I'm a little afraid of comparability issues with some gems.

**Comment by Sammidysam** [comment]
If only a darker navbar is wanted, then we can just customize the Bootstrap variables---there would be no reason to switch the style sheet language in that case.  I guess whether we need to look into LESS more or not is determined by how much Bootstrap we want to change.  If it is a _lot_, then it is probably better to move towards LESS because then we can let Bootswatch take care of most of the changes.  If it's just a few colors or a few components here and there, then we can just set up a `bootstrap-custom.scss` file and work with it.

In that situation, we would go back to using the Rails comments in the `application.css.scss` file that were removed in e0c73bdc4458ca4493bed01a7175c27de631a73b.  Then, we would make a separate file called `bootstrap-custom.scss` which would be where we would `@import "bootstrap";` Bootstrap and customize it.  It would be included by the application _after_ `application.css.scss`.  We would have Bootstrap-related customizations in that file.  General customizations would still go in `application.css.scss`.  I am pretty sure that this would work; I think that I did this in something that I made a really long time ago but I believe that I lost the source code.  Nonetheless, I think that that would work for using Sass to customize bootstrap.

Anyway, using LESS may work, but possibly using SCSS is safer.  ~~There are definitely LESS to SCSS converters out there, which could ease the work of translating files.~~  I do not think there are many, if any, converters out there, but all that would really need to be done is a few find/replaces and then a few fixing of features that differ.  There aren't many.  To quote [cimmanon](http://stackoverflow.com/users/1652962/cimmanon) on Stack Overflow, "The only LESS -> Sass port that's difficult to do with a quick search/replace is mixin invocation because classes are mixins in LESS."  I feel like it may suffice to customize the Bootstrap ourselves, though.

In conclusion, I think that sticking with SCSS might be good, but translating the Bootswatch to SCSS if necessary or just customizing the Bootstrap within the SCSS will need to be done.  If we need to go to LESS, it would be okay.

**Comment by rye** [comment]
So basically, if we're using themes, the only way I can see it working out is if we use less, since we can't get a direct SCSS file from bootswatch.

Or, we just customize it (what I consider to be the better option since it's the only one we can seem to agree on at this point).

**Comment by Sammidysam** [comment]
We can translate LESS to SCSS fairly easily.  In my third paragraph, I said, "I do not think there are many, if any, converters out there, but all that would really need to be done is a few find/replaces and then a few fixing of features that differ. There aren't many. To quote cimmanon on Stack Overflow, 'The only LESS -> Sass port that's difficult to do with a quick search/replace is mixin invocation because classes are mixins in LESS.'"

**Comment by Sammidysam** [comment]
Also, you mentioned this earlier, but [bootswatch-scss](https://github.com/log0ymxm/bootswatch-scss) contains files that we can copy over if we don't use Yeti.  If we do, then we should probably submit a pull request to that repository after we convert it, if we do translate files.

**Comment by cg505** [comment]
What I think we should do:

Download the less for Yeti.
Run the converter on it.
Take a quick lookthrough by hand to make sure everything got translated and nothing is super screwy.
Test. Fix. Repeat.

Obviously this would be done on a branch. It really doesn't seem like a ton of work.

**Comment by cg505** [comment]
By "the converter" I mean the converter in the bootstrap-sass repo.

**Comment by cg505** [comment]
Also, if we just want a dark navbar, we can literally just use `.navbar-inverse`.

**Comment by Sammidysam** [comment]
Are we going to use Bootswatch Yeti then, and go through that process, or just deal with the default bootstrap?

Also, if we translate the LESS, note that we should make a file in our style sheets directory called `yeti-bootstrap.scss` or `bootstrap-yeti.scss` in which the format is as follows.

```
Customized Bootstrap Variables
example:
$navbar-height: 60px;

@import "bootstrap";

Bootstrap overrides
```

It will then automatically get included by our program by being in the style sheets directory.

**Comment by rye** [comment]
I would say just normal bootstrap with extra configurations if necessary.

**Comment by Sammidysam** [comment]
But Yeti is still ideal?

**Comment by rye** [comment]
If we can get it without doing conversion (i.e. using less or raw files)
yes, if not no.

// Kristofer "four04" Rye

**Comment by Sammidysam** [comment]
I feel like both using LESS or CSS files would be more work than converting LESS to SCSS.  We _can_ switch to using LESS, but it won't be as easy as converting.  Though if converting files sounds that horrible, I guess we can look at LESS.

**Comment by rye** [comment]
I repeat: We do not need to "switch" to less, just figure out how to use
less and SCSS simultaneously.

**Comment by Sammidysam** [comment]
That sounds hard.

**Comment by cg505** [comment]
It sounds hard, but it may actually not be very hard at all.
http://shcatula.wordpress.com/2013/05/06/mixin-less-sass-with-twitter-bootstrap-rails/

**Comment by Sammidysam** [comment]
A comment on that includes "looking at your source figured that this will not work with rails 4, because of sass-rails importer changes."  It may not work.

**Comment by rye** [comment]
So I vote that we just stick to tweaking default bootstrap (change navbar to navbar-inverse). I'll delete the `theme` branch and re-create it, with different changes.

**Comment by rye** [comment]
Or just revert all of the commits in it (This is the alternative I'll probably go with.)

**Comment by cg505** [comment]
Why would we do that?

**Comment by rye** [comment]
I've deleted it, and because we don't need the commits that were made on that branch.

**Comment by cg505** [comment]
Okay. I was wondering why reverting would be preferable.

**Comment by Sammidysam** [comment]
Sounds good.  It is good that we finally decided what to do.

**Comment by Sammidysam** [comment]
Shall we close this now?

**Comment by rye** [comment]
Yeah, that's fine.
